api-extractor/src/Extractor.ts

```diff
@@ -49,25 +58,28 @@ export default class Analyzer {
    */
   public docItemLoader: DocItemLoader;
 
+  private _compilerOptions: ts.CompilerOptions;
+
   /**
     * The default implementation of ApiErrorHandler, which merely writes to console.log().
     */
   public static defaultErrorHandler(message: string, fileName: string, lineNumber: number): void {
     console.log(`ERROR: [${fileName}:${lineNumber}] ${message}`);
   }
 
-  constructor(errorHandler?: ApiErrorHandler) {
-    this.errorHandler = errorHandler || Analyzer.defaultErrorHandler;
+  constructor(options: IExtractorOptions) {
+    this._compilerOptions = options.compilerOptions;
+    this.docItemLoader = new DocItemLoader(options.compilerOptions.rootDir);
+    this.errorHandler = options.errorHandler || Extractor.defaultErrorHandler;
   }
 
   /**
     * Analyzes the specified project.
     */
   public analyze(options: IApiAnalyzerOptions): void {
```

Comment:
If you're keeping the name analyze(), then the interface should be something like IExtractorAnalyzeOptions